Expensify/App
# Deselect attendees when splitting bill

## Progress so far.

- [x] Toggle between selected and unselected participants.
- [x] The amount is split between selected participants only.
- [x] Disable button when no participants is selected. 
- [x] New group shouldn't be created for selected participants.
- [x] (**This will be handled by API layer. [See Reference](https://github.com/Expensify/Expensify.cash/issues/3841#issuecomment-875005197)**) Add the split comment to the group DM with selected participants only.


### Details

[Comment ](https://github.com/Expensify/Expensify.cash/issues/3841#issuecomment-872567843)
To check if a split happening from a group DM, `props.route.params.reportID` is checked.


### Fixed Issues

$ #3841 

### Tests / QA

1. Login to E.cash
2. Go to a group DM
3. Click on +, then split bill.
4. Enter amount and go next.
5. Verify that you can unselect 'who was there'.
6. Verify that the amount is split correctly.
6. After splitting bill. Verify that bill is split only among the selected participants.
7. Verify that no new group is created.
8. Verify that split comment in group DM only lists the selected participants.

### Tested On
Don't have a Mac to test iOS, Desktop.

### Screenshots


#### Web



https://user-images.githubusercontent.com/29673073/125373585-190bd880-e38e-11eb-9587-561fb6826e57.mp4


#### Mobile Web

<img src="https://user-images.githubusercontent.com/29673073/125374890-aa7c4a00-e390-11eb-92e5-ddb80030be2c.jpeg" height="500">


#### Desktop


#### iOS


#### Android


https://user-images.githubusercontent.com/29673073/125374691-32158900-e390-11eb-98b5-77241d3d396c.mp4
